src/Validator/Constraints/BonusPointsApplyValidator.php

```diff
@@ -20,70 +14,28 @@ final class BonusPointsApplyValidator extends ConstraintValidator
     /** @var BonusPointsStrategyRepositoryInterface */
     private $bonusPointsStrategyRepository;
 
-    /** @var BonusPointsStrategyRulesEligibilityChecker */
-    private $bonusPointsStrategyEligibilityChecker;
-
-    /** @var CartContextInterface */
-    private $cartContext;
-
-    public function __construct(
-        BonusPointsStrategyRepositoryInterface $bonusPointsStrategyRepository,
-        BonusPointsStrategyEligibilityCheckerInterface $bonusPointsStrategyEligibilityChecker,
-        CartContextInterface $cartContext
-    ) {
+    public function __construct(BonusPointsStrategyRepositoryInterface $bonusPointsStrategyRepository)
+    {
         $this->bonusPointsStrategyRepository = $bonusPointsStrategyRepository;
-        $this->bonusPointsStrategyEligibilityChecker = $bonusPointsStrategyEligibilityChecker;
-        $this->cartContext = $cartContext;
     }
 
     public function validate($bonusPoints, Constraint $constraint): void
     {
-        $bonusPointsStrategies = $this->bonusPointsStrategyRepository->findActiveByCalculatorType(PerOrderPriceCalculator::TYPE);
-
-        if (\count($bonusPointsStrategies) === 0) {
+        if (null === $bonusPoints) {
             return;
         }
 
-        $eligibleBonusPointsStrategies = $this->extractOnlyEligibleStrategies($bonusPointsStrategies);
-
-        if (\count($eligibleBonusPointsStrategies) === 0) {
-            $this->context->getViolations()->remove(0);
-
-            $this->context->buildViolation($constraint->noProductsFromBonusPointsStrategyTaxonMessage)->addViolation();
+        $bonusPointsStrategies = $this->bonusPointsStrategyRepository->findActiveByCalculatorType(PerOrderPriceCalculator::TYPE);
 
+        if (\count($bonusPointsStrategies) === 0) {
             return;
         }
 
-        foreach ($eligibleBonusPointsStrategies as $eligibleBonusPointsStrategy) {
+        if ($bonusPoints % 100 !== 0 || $bonusPoints < 100) {
             $this->context->getViolations()->remove(0);
+            $this->context->buildViolation($constraint->invalidBonusPointsValueMessage)->addViolation();
```

Comment:
are there any integration tests that check this validation?

Reply:
Yes, of course :)
```
@ui @javascript
    Scenario: Unsuccessfully using of awarded bonus points with "per order price" calculator
        When I want to use "0.32" bonus points
        Then I should be notified that this value should be natural number, greater than or equal to 1
```